### ISSUE TITLE
ImageSubdomain meshmodifier is processed against the NON-refined mesh

### ISSUE THREAD

**Issue by mlesueur** [idaholab/moose#7584]
### Description of the error report
When using the [ImageSubdomain](https://github.com/idaholab/moose/blob/devel/framework/src/meshmodifiers/ImageSubdomain.C) meshmodifier, the mesh is compared to the image data, and we have the creation of two subdomains.
More precisely we are looking if the image's pixel value of the centre of the element is corresponding to one block or the other.

The problem is that this comparison is done on the NON-refined mesh!
In a mesh where resolution is defined by nx,ny,nz and uniform_refine, this meshmodifier is processed with only nx,ny,nz!

### Information for reproducing the error
Let us modify slightly the input file formulation of the [ImageSubdomain test](https://github.com/idaholab/moose/blob/devel/test/tests/mesh_modifiers/image_subdomain/image_2d.i):

- In the test file: nx=**176**, ny=**287**, uniform_refine=0
- In the failed test file: nx=11, ny=18, uniform_refine=4

->nx'=11x2^4=**176**, ny'=18x2^4=**288**
It means that these two files should give the same result. However these are the screenshots of the two outputs:
![image](https://cloud.githubusercontent.com/assets/11986570/17798825/69a84600-6618-11e6-9d36-588e556498c7.png)
![image](https://cloud.githubusercontent.com/assets/11986570/17798831/7a898c54-6618-11e6-8297-1d307a3d7a39.png)

### Identified impact
This constraint means that when we are using adaptivity after this meshmodifier, we would not be able to coarsen any elements, because an element can only be coarsened if it has been previously refined.

**Comment by YaqiWang** [comment]
Yes, mesh modifiers are executed before uniform refinement. ImageSubdomain is an exception in that it depends on the uniform refinement. But that can be fixed easily by adding a parameter in MeshModifier like `post_uniform_refinement`.

**Comment by permcody** [comment]
Yaqi maybe right here. If that simple fix will work for you, we might try that as well. Take a look at the email I composed this afternoon too.
